frappe/model/base_document.py

```diff
@@ -564,10 +564,12 @@ def get_msg(df, docname):
 				if not frappe.get_meta(doctype).get('is_virtual'):
 					if not fields_to_fetch:
 						# cache a single value type
-						values = frappe._dict(name=frappe.db.get_value(doctype, docname,
-							'name', cache=True))
+						values = frappe.db.get_value(doctype, docname,
+							'name', cache=True, as_dict = True)
+						if not values:
+							values = frappe._dict(name=None)
```

Comment:
In my early dev work it was actually causing LinkValidationErrors in the old form - but I just tested now and it seems one of my other commits fixed that issue, so this can be reverted.